### PR TITLE
Fix NBT issue & Fix Cake World Incompat

### PR DESCRIPTION
**Pull request by sleepingdragoninn**
using [reference](https://github.com/sisby-folk/switchy/blob/1.18/src/main/java/folk/sisby/switchy/mixin/MixinPlayerEntity.java#L27)

### CODE REVIEW

**Comment by sleepingdragoninn** [comment]
Added two small additional if-guards to the blockentity, ensuring that C2S packets are only sent if the blockentity is in the same world as the player. This fixes compatibility with Cake World.

**Comment by sylv256** [comment]
i forgor to only send S2C and C2S packets if it's in the same world lol

File: gradle.properties [hunk 1]
```diff
@@ -2,6 +2,6 @@ kotlin.code.style=official
 org.gradle.jvmargs=-Xmx1G
 
 # Mod Properties
-modVersion = 0.1.3-1.19.2+MODFEST.4
+modVersion = 0.1.4-1.19.2+MODFEST.5
```
Comment (sylv256) [inline gradle.properties:5]: yes thank you

**Review by sylv256 — APPROVED** [review]
yes

**Comment by maximumpower55** [comment]
:tiny_tater_tot:

**Pull request merged**